src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -7420,8 +7420,11 @@ public override BoundNode VisitAwaitExpression(BoundAwaitExpression node)
             }
             else
             {
-                // Update method based on inferred receiver type: see https://github.com/dotnet/roslyn/issues/29605.
-                SetResultType(node, awaitableInfo.GetResult.ReturnTypeWithAnnotations.ToTypeWithState());
+                var reinferredGetResult = _visitResult.RValueType.Type is NamedTypeSymbol taskAwaiterType
+                    ? awaitableInfo.GetResult.OriginalDefinition.AsMember(taskAwaiterType)
```

Comment:
>awaitableInfo.GetResult.OriginalDefinition.AsMember(taskAwaiterType) [](start = 22, length = 68)

Just curious: Why use `NamedTypeSymbol.AsMember()` rather than `AsMemberOfType(TypeSymbol, Symbol)` as used in other cases in `NullableWalker`?

Reply:
Hmm, I must have missed it when I went looking for it, but it seems like `NullableWalker.AsMemberOfType` is a more complex/general purpose method, and it eventually calls down into `MethodSymbol.AsMember(NamedTypeSymbol)` which I use directly here.